src/Searchlight/SqlExecutor.cs

```diff
@@ -331,6 +333,25 @@ private static string DecorateTableName(SqlDialect dialect, string tableName, Se
 
             return tableName;
         }
+
+        private static string GetColumnNameOrJsonValue(SqlDialect dialect, ColumnInfo columnInfo, string[] jsonKeys)
+        {
+            var columnName = columnInfo?.OriginalName ?? string.Empty;
+            var isJson = columnInfo?.IsJson ?? false;
+            jsonKeys = jsonKeys ?? Array.Empty<string>();
+
+            switch (dialect)
+            {
+                case SqlDialect.MicrosoftSqlServer:
```

Comment:
This might be easier to read as a switch expression

pseudo-code:
```csharp
return dialect switch =>
  SqlDialect.MicrosoftSqlServer when !isJson: ...
  SqlDialect.MicrosoftSqlServer when isJson: ...
etc
```

Reply:
The Searchlight project is on C# lang version 7.3, I don't think switch expressions are supported. We could upgrade the C# version though, I agree this would be much cleaner and easier to read

Reply:
Got it. We probably shouldn't upgrade the langversion in our fork.